src/cryptonote_core/cryptonote_core.cpp

```diff
@@ -2509,4 +2489,19 @@ namespace cryptonote
   {
     raise(SIGTERM);
   }
+  //-----------------------------------------------------------------------------------------------
+  bool core::find_nonce_for_given_block(const get_block_hash_t &gbh, block &bl, const difficulty_type &diffic, uint64_t height)
```

Comment:
This function has no dependencies on core, it'd be better off in cryptonote_basic.h

Reply:
Yea I couldn't figure out the best place for this one, basic seems like a better option